central/hash/manager/deduper.go

```diff
@@ -79,40 +79,91 @@ func skipDedupe(msg *central.MsgFromSensor) bool {
+		if alert.IsRuntimeAlertResult(msg.GetEvent().GetAlertResults()) {
+			return true
+		}
+		// This can occur for a very short-lived deployment where alerts are not generated
+		// but the deployment is being removed
+		if alert.IsAlertResultResolved(msg.GetEvent().GetAlertResults()) {
+			return true
+		}
+		if alert.AnyAttemptedAlert(msg.GetEvent().GetAlertResults().GetAlerts()...) {
+			return true
+		}
+		return false
+
+	// Network and Security Resources
+	case *central.SensorEvent_NetworkPolicy:
+		return false
+	case *central.SensorEvent_Secret:
+		return false
+
+	// Infrastructure Resources
+	case *central.SensorEvent_Node:
 		return true
-	}
-	// Autogenerated image integrations currently do not have consistent ID
-	if eventMsg.Event.GetImageIntegration() != nil {
+	case *central.SensorEvent_NodeInventory:
 		return true
-	}
-	if alert.IsRuntimeAlertResult(msg.GetEvent().GetAlertResults()) {
+	case *central.SensorEvent_IndexReport:
 		return true
-	}
-	// This can occur for a very short-lived deployment where alerts are not generated
-	// but the deployment is being removed
-	if alert.IsAlertResultResolved(msg.GetEvent().GetAlertResults()) {
+
+	// RBAC Resources
+	case *central.SensorEvent_ServiceAccount:
+		return false
+	case *central.SensorEvent_Role:
+		return false
+	case *central.SensorEvent_Binding:
+		return false
+
+	// Process and Runtime
+	case *central.SensorEvent_ProcessIndicator:
 		return true
-	}
-	if alert.AnyAttemptedAlert(msg.GetEvent().GetAlertResults().GetAlerts()...) {
+	case *central.SensorEvent_ReprocessDeployment:
 		return true
-	}
-	if eventMsg.Event.GetReprocessDeployment() != nil {
+
+	// Metadata and Configuration
+	case *central.SensorEvent_ProviderMetadata:
 		return true
-	}
-	if eventMsg.Event.GetIndexReport() != nil {
+	case *central.SensorEvent_OrchestratorMetadata:
 		return true
-	}
-	if eventMsg.Event.GetNodeInventory() != nil {
+	case *central.SensorEvent_ImageIntegration:
 		return true
-	}
-	if eventMsg.Event.GetVirtualMachine() != nil {
+
+	// Compliance Operator (V1)
+	case *central.SensorEvent_ComplianceOperatorResult,
+		*central.SensorEvent_ComplianceOperatorProfile,
+		*central.SensorEvent_ComplianceOperatorRule,
+		*central.SensorEvent_ComplianceOperatorScanSettingBinding,
+		*central.SensorEvent_ComplianceOperatorScan:
 		return true
-	}
-	if eventMsg.Event.GetVirtualMachineIndexReport() != nil {
+
+	// Virtual Machines
+	case *central.SensorEvent_VirtualMachineIndexReport, *central.SensorEvent_VirtualMachine:
+		return true
+
+	// Compliance Operator V2
+	case *central.SensorEvent_ComplianceOperatorResultV2,
+		*central.SensorEvent_ComplianceOperatorProfileV2,
+		*central.SensorEvent_ComplianceOperatorRuleV2,
+		*central.SensorEvent_ComplianceOperatorScanV2,
+		*central.SensorEvent_ComplianceOperatorScanSettingBindingV2,
+		*central.SensorEvent_ComplianceOperatorSuiteV2,
+		*central.SensorEvent_ComplianceOperatorRemediationV2:
+		return true
+
+	default:
 		return true
```

Comment:
Maybe we should panic (in development builds) or at least log something here. Otherwise I feel new events might not be added here explicitly. Maybe we could also add a linter so that way any new event needs to be explicitly added here.